File: app/src/main/java/com/syki/fidoreader/fagment/ReaderButtonFragment.java
package com.syki.fidoreader.fagment;

import androidx.lifecycle.ViewModelProviders;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.syki.fidoreader.MainActivity;
import com.syki.fidoreader.R;

public class ReaderButtonFragment extends Fragment {

    MainActivity mainActivity;
    private ReaderButtonViewModel mViewModel;

    public static ReaderButtonFragment newInstance() {
        return new ReaderButtonFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.reader_button_fragment, container, false);

        Button btn = (Button)view.findViewById(R.id.readerActivationBtn);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mainActivity.onChangeFragment();
            }
        });

        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mainActivity = (MainActivity)getActivity();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mainActivity = null;
    }



}

File: app/src/main/java/com/challenge/fidoreader/fagment/ReaderButtonFragment.kt
package com.challenge.fidoreader.fagment

import android.os.Bundle
import android.util.Log
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.Fragment
import com.challenge.fidoreader.MainActivity
import com.challenge.fidoreader.R
import java.lang.Exception

class ReaderButtonFragment : Fragment(){
    val TAG = "ReaderButtonFragment"

    lateinit var imageView: ImageView
    lateinit var textView2: TextView
    lateinit var textView3: TextView

    lateinit var btn:Button
    lateinit var getinfoBtn:Button
    lateinit var clientPinBtn:Button

    var hasClientPIN = false

    lateinit var mainActivity:MainActivity

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.v(TAG, "onCreateView")

        var mainView: View = inflater.inflate(R.layout.fragment_reader_button, container, false)

        btn = mainView.findViewById(R.id.readerActivationBtn)
        btn.setOnClickListener{
            mainActivity.onChangeFragmentToList()
        }

        btn.isEnabled = false

        getinfoBtn = mainView.findViewById(R.id.KonaBIOPASSGetInfoBtn)
        clientPinBtn = mainView.findViewById(R.id.KonaBIOPASSPINBtn)
        clientPinBtn.isEnabled = false

        getinfoBtn.setOnClickListener{
            try {
                initTable(mainView)
                if(hasClientPIN){
                    clientPinBtn.text = "Change PIN"
                }else{
                    clientPinBtn.text = "Set PIN"
                }
            }catch (e: Exception){
                e.printStackTrace()
            }
        }
        getinfoBtn.isEnabled = false

        clientPinBtn.setOnClickListener {
            var pinFragment:DialogFragment? = null
        }


        return mainView
    }

    fun initTable(view: View){
        var getInfoTable: TableLayout = view.findViewById(R.id.KonaBIOPASSGetInfoTable)
        var row:TableRow = TableRow(activity)
        var layoutParams:TableRow.LayoutParams = TableRow.LayoutParams(TableRow.LayoutParams.WRAP_CONTENT)
        row.layoutParams = layoutParams

        var funtionRow: TextView = TextView(activity)
        funtionRow.gravity = Gravity.CENTER
        funtionRow.text = "기능"
        row.addView(funtionRow)

        var supported:TextView = TextView(activity)
        supported.gravity = Gravity.CENTER
        supported.text = "값"
        row.addView(supported)

        getInfoTable.addView(row)

        getInfoPrint(getInfoTable)
    }

    fun getInfoPrint(getInfoTable: TableLayout) {

    }


}



